### ISSUE TITLE
[Damla, Ceylin] Fix Web Search & Minor Errors & Refactoring

### ISSUE THREAD

**Issue by damlakayikci** [bounswe/bounswe2024group9#258]
#### The web search bar doesn't function as desired and there are some other errors.

* The search bar doesn't show the text typed.
* Searching doesn't function.
* The questions on feed page are static.
* Some parts of the page (nav bar, side bars) must be consolidated into a single file to avoid duplicate code.

Tasks
* [ ] Add functionality to search bar
* [ ] Create PageComponents.js and move the necessary parts here
* [ ] Pull questions from the database
* [ ] Redirect search results to the corresponding language page. 

Deadline: 17.11.24
Estimated effort: 10 hours
Reviewer: @serhan-cakmak @ErayEroglu

**Comment by damlakayikci** [comment]
Hey, I finished PageComponents.js. Would love to hear your feedback about it! There is still some duplicate code between Feed.js and SearchResult.js but do you think I should also move the functions that are used in the navbar to somewhere else?

**Comment by ceylingebes** [comment]
PageComponents.js looks cool. I think we should also move the duplicated functions to the PageComponents since we can use navbar in upcoming pages such as profile

**Comment by ceylingebes** [comment]
Also I think we completed task #4 -redirecting search results- as well, aren't we? I checked it but feel free to uncheck it if I'm wrong Damla. Nice work

**Comment by Mutti499** [comment]
Okay seems like a dope front end! However, I'm starting to feel like I'm hindering your efforts by changing URLs and views completely. Can we talk about them tomorrow? Because we will have to make one more update after them I guess. Maybe I am missing and there are no problems but I have anxiety.. UPSIE.

**Comment by ceylingebes** [comment]
After loong conversations, we have updated all urls according to the new version of the backend therefore we are ready for a review @serhan-cakmak @ErayEroglu can you check if everything is ok?

**Comment by ceylingebes** [comment]
Are we talking about web-search branch? If so, I think we merged that into main in Lab 6 PR so I'm not sure we need to open another PR. What do you think? @serhan-cakmak @Mutti499

**Comment by damlakayikci** [comment]
this branch is currently not ready, but since it is a lower priority than other tasks I had to move to other branches to deal with more important problems. I'll let you know once it is ready to be merged. Thanks for your patience!
